### PR TITLE
Move column to the very first place after freezing

### LINKED ISSUE #5: 1.4 Column pinning (freeze)
Depends on https://github.com/vaadin/px-data-grid/issues/4

Click on `Pin column` sets the `frozen` property of corresponding `vaadin-grid-column` to `true`

For pinned column: 
 - `Pin column` menu item in the dropdown menu should be hidden
 - `Unpin column` menu item should be shown

Click on `Unpin column` sets the `frozen` property of corresponding `vaadin-grid-column` to `false`

### UI specs

image

image

image

### PR DESCRIPTION
**Pull request by Toshegg**
1) Move column to the very first place after freezing
2) Move column to lastFrozen + 1 position after unfreezing
3) Fix px-dropdown overlay issue

Fixes #5

### CODE REVIEW

File: px-data-grid-header-cell.html [hunk 1]
```diff
@@ -142,6 +158,12 @@
           if (!dropdownOpened && !this._mouseover) {
             this._dropdownHidden = true;
           }
+
+          if (dropdownOpened) {
+            this.parentElement.assignedSlot.parentElement.style.zIndex = '3';
```
Comment (limonte) [inline px-data-grid-header-cell.html]: 1. DRY: assign the variable before if statement
2. add the explanatory comment why this is needed

Comment (Toshegg) [reply px-data-grid-header-cell.html]: Done.

**Comment by alump** [comment]
One request... the number one should be

1. Move column first if no frozen columns, or move it left side of right most frozen column.

**Comment by Toshegg** [comment]
@alump, is it requested, or why it should behave like so?
For example, if we take a look at the [ag-grid](https://www.ag-grid.com/example.php), it places pinned column to the very first place, what is pretty convenient and also much easier to implement.

**Comment by alump** [comment]
Because if it doesn't work this way, then you want to freeze multiple orders you would need to do it in reverse order you want those to be frozen.

Anyway, which option is easier, do that.

**Pull request merged**